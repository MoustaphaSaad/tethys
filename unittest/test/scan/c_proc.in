constant msg "Hello, World!\0"

proc C.add(i32, i32)

proc main
	halt	
end