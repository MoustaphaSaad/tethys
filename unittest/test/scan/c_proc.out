line: 1, col: 0, kind: "CONSTANT" str: "constant"
line: 1, col: 9, kind: "<ID>" str: "msg"
line: 1, col: 13, kind: "<STRING>" str: "Hello, World!\0"
line: 3, col: 1, kind: "PROC" str: "proc"
line: 3, col: 6, kind: "<ID>" str: "C.add"
line: 3, col: 11, kind: "(" str: "("
line: 3, col: 12, kind: "i32" str: "i32"
line: 3, col: 15, kind: "," str: ","
line: 3, col: 17, kind: "i32" str: "i32"
line: 3, col: 20, kind: ")" str: ")"
line: 5, col: 1, kind: "PROC" str: "proc"
line: 5, col: 6, kind: "<ID>" str: "main"
line: 6, col: 2, kind: "HALT" str: "halt"
line: 7, col: 1, kind: "END" str: "end"
